barman/utils.py

```diff
@@ -775,3 +777,47 @@ def check_size(value):
     if int_value is None or int_value < 1:
         raise ArgumentTypeError("'%s' is not a valid size string" % value)
     return int_value
+
+
+def check_backup_name(backup_name):
+    # TODO check the backup name does not match a backup ID and is otherwise not
+    # illegal (whatever that means)
+    if backup_name is None:
+        raise ArgumentTypeError("Backup name cannot be None")
+    if backup_name == "":
+        raise ArgumentTypeError("Backup name cannot be empty")
+    if is_backup_id(backup_name):
+        raise ArgumentTypeError(
+            "Backup name '%s' is not allowed: backup ID" % backup_name
+        )
+    if backup_name in (RESERVED_BACKUP_IDS):
+        raise ArgumentTypeError(
+            "Backup name '%s' is not allowed: reserved word" % backup_name
+        )
+    return backup_name
+
+
+def is_backup_id(backup_id):
+    # TODO check if backup_id matches pattern (is this logic somewhere already?)
+    return bool(re.match(r"(\d{8})T\d{6}$", backup_id))
```

Comment:
I have a couple of questions regarding this check:
1. It's pretty weak. Should we instead be verifying it is a parseable short-form ISO 8601 timestamp?
2. Has the backup ID historically always been short-form ISO 8601? Is it possible a backup ID could be valid but not pass this check?